src/test/java/com/google/maps/PlacesApiTest.java

```diff
@@ -0,0 +1,540 @@
+ * permissions and limitations under the License.
+ */
+
+package com.google.maps;
+
+import com.google.maps.model.AddressComponentType;
+import com.google.maps.model.LatLng;
+import com.google.maps.model.OpeningHours.Period;
+import com.google.maps.model.OpeningHours.Period.OpenClose.DayOfWeek;
+import com.google.maps.model.Photo;
+import com.google.maps.model.PlaceDetails;
+import com.google.maps.model.PlaceDetails.PriceLevel;
+import com.google.maps.model.PlaceDetails.Review.AspectRating.RatingType;
+import com.google.maps.model.PlaceIdScope;
+import com.google.maps.model.PlacesSearchResponse;
+import com.google.maps.model.PlacesSearchResult;
+import com.google.maps.model.QueryAutocompletePrediction;
+import com.google.mockwebserver.MockResponse;
+import com.google.mockwebserver.MockWebServer;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.junit.Test;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.List;
+import java.util.Scanner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class PlacesApiTest {
+
+  public static final String GOOGLE_SYDNEY = "ChIJN1t_tDeuEmsRUsoyG83frY4";
+  public static final String QUAY_PLACE_ID = "ChIJ02qnq0KuEmsRHUJF4zo1x4I";
+  public static final String QUERY_AUTOCOMPLETE_INPUT = "pizza near par";
+
+  private final GeoApiContext context = new GeoApiContext().setApiKey("AIzaFakeKey");
+  private final String placeDetailResponseBody;
+  private final String quayResponseBody;
+  private final String queryAutocompleteResponseBody;
+  private final String queryAutocompleteWithPlaceIdResponseBody;
+  private final String textSearchResponseBody;
+  private final String textSearchPizzaInNYCbody;
+
+  public PlacesApiTest() {
+    placeDetailResponseBody = retrieveBody("PlaceDetailsResponse.json");
+    quayResponseBody = retrieveBody("PlaceDetailsQuay.json");
+    queryAutocompleteResponseBody = retrieveBody("QueryAutocompleteResponse.json");
+    queryAutocompleteWithPlaceIdResponseBody = retrieveBody("QueryAutocompleteResponseWithPlaceID.json");
+    textSearchResponseBody = retrieveBody("TextSearchResponse.json");
+    textSearchPizzaInNYCbody = retrieveBody("TextSearchPizzaInNYC.json");
+  }
+
+  private String retrieveBody(String filename) {
+    InputStream input = this.getClass().getResourceAsStream(filename);
+    Scanner s = new java.util.Scanner(input).useDelimiter("\\A");
+    String body = s.next();
+    if (body == null || body.length() == 0) {
+      throw new IllegalArgumentException("filename '" + filename + "' resulted in null or empty body");
+    }
+    return body;
+  }
+
+  @Test
+  public void testPlaceDetailsRequest() throws Exception {
+    MockResponse response = new MockResponse();
+    response.setBody("");
+    MockWebServer server = new MockWebServer();
```

Comment:
Move the server to the class level (@BeforeClass / @Before, whatever it is), along with the context.setBaseUrl...